include/libopencm3/stm32/f7/rcc.h

```diff
@@ -518,94 +604,164 @@
-#define RCC_DCKCFGR1_PLLSAIDIVQ_MASK		0x1f
+/**@}*/
+/* RCC_DCKCFGR1[12:8]: PLLSAIDIVQ */
 #define RCC_DCKCFGR1_PLLSAIDIVQ_SHIFT		8
-#define RCC_DCKCFGR1_PLLI2SDIVQ_MASK		0x1f
+#define RCC_DCKCFGR1_PLLSAIDIVQ_MASK		0x1f
+/* RCC_DCKCFGR1[4:0]: PLLI2SDIVQ */
 #define RCC_DCKCFGR1_PLLI2SDIVQ_SHIFT		0
+#define RCC_DCKCFGR1_PLLI2SDIVQ_MASK		0x1f
 
 /* --- RCC_DCKCFGR2 values -------------------------------------------------- */
 
-#define RCC_DCKCFGR2_DSISEL			(1<<28)
-#define RCC_DCKCFGR2_SDMMCSEL			(1<<28)
+#define RCC_DCKCFGR2_DSISEL			(1<<30)
+#define RCC_DCKCFGR2_SDMMC2SEL			(1<<29)
+#define RCC_DCKCFGR2_SDMMC1SEL			(1<<28)
 #define RCC_DCKCFGR2_CK48MSEL			(1<<27)
 #define RCC_DCKCFGR2_CECSEL			(1<<26)
-#define RCC_DCKCFGR2_LPTIM1SEL_MASK		0x3
+/* RCC_DCKCFGR2[25:24]: LPTIM1SEL */
 #define RCC_DCKCFGR2_LPTIM1SEL_SHIFT		24
-#define RCC_DCKCFGR2_I2C4SEL_MASK		0x3
+#define RCC_DCKCFGR2_LPTIM1SEL_MASK		0x3
+/* RCC_DCKCFGR2[23:22]: I2C4SEL */
 #define RCC_DCKCFGR2_I2C4SEL_SHIFT		22
-#define RCC_DCKCFGR2_I2C3SEL_MASK		0x3
+#define RCC_DCKCFGR2_I2C4SEL_MASK		0x3
+/* RCC_DCKCFGR2[21:20]: I2C3SEL */
 #define RCC_DCKCFGR2_I2C3SEL_SHIFT		20
-#define RCC_DCKCFGR2_I2C2SEL_MASK		0x3
+#define RCC_DCKCFGR2_I2C3SEL_MASK		0x3
+/* RCC_DCKCFGR2[19:18]: I2C2SEL */
 #define RCC_DCKCFGR2_I2C2SEL_SHIFT		18
-#define RCC_DCKCFGR2_I2C1SEL_MASK		0x3
+#define RCC_DCKCFGR2_I2C2SEL_MASK		0x3
+/* RCC_DCKCFGR2[17:16]: I2C1SEL */
 #define RCC_DCKCFGR2_I2C1SEL_SHIFT		16
-#define RCC_DCKCFGR2_UART8SEL_MASK		0x3
+#define RCC_DCKCFGR2_I2C1SEL_MASK		0x3
+/* RCC_DCKCFGR2[15:14]: UART8SEL */
 #define RCC_DCKCFGR2_UART8SEL_SHIFT		14
-#define RCC_DCKCFGR2_UART7SEL_MASK		0x3
+#define RCC_DCKCFGR2_UART8SEL_MASK		0x3
+/* RCC_DCKCFGR2[13:12]: UART7SEL */
 #define RCC_DCKCFGR2_UART7SEL_SHIFT		12
-#define RCC_DCKCFGR2_USART6SEL_MASK		0x3
+#define RCC_DCKCFGR2_UART7SEL_MASK		0x3
+/* RCC_DCKCFGR2[11:10]: USART6SEL */
 #define RCC_DCKCFGR2_USART6SEL_SHIFT		10
-#define RCC_DCKCFGR2_UART5SEL_MASK		0x3
+#define RCC_DCKCFGR2_USART6SEL_MASK		0x3
+/* RCC_DCKCFGR2[9:8]: UART5SEL */
 #define RCC_DCKCFGR2_UART5SEL_SHIFT		8
-#define RCC_DCKCFGR2_UART4SEL_MASK		0x3
+#define RCC_DCKCFGR2_UART5SEL_MASK		0x3
+/* RCC_DCKCFGR2[7:6]: UART4SEL */
 #define RCC_DCKCFGR2_UART4SEL_SHIFT		6
-#define RCC_DCKCFGR2_UART3SEL_MASK		0x3
+#define RCC_DCKCFGR2_UART4SEL_MASK		0x3
+/* RCC_DCKCFGR2[5:4]: UART3SEL */
 #define RCC_DCKCFGR2_UART3SEL_SHIFT		4
-#define RCC_DCKCFGR2_UART2SEL_MASK		0x3
+#define RCC_DCKCFGR2_UART3SEL_MASK		0x3
+/* RCC_DCKCFGR2[3:2]: UART2SEL */
 #define RCC_DCKCFGR2_UART2SEL_SHIFT		2
-#define RCC_DCKCFGR2_UART1SEL_MASK		0x3
+#define RCC_DCKCFGR2_UART2SEL_MASK		0x3
+/* RCC_DCKCFGR2[1:0]: UART1SEL */
 #define RCC_DCKCFGR2_UART1SEL_SHIFT		0
+#define RCC_DCKCFGR2_UART1SEL_MASK		0x3
 
+/* PLLSAI1 helper macros */
+static inline void rcc_pllsai_enable(void)
+{
+	RCC_CR |= RCC_CR_PLLSAION;
+}
+
+static inline bool rcc_pllsai_ready(void)
+{
+	return (RCC_CR & RCC_CR_PLLSAIRDY) != 0;
+}
```

Comment:
we've not done this for other things, I don't want to start doing this now.  Just put them in the c files like every other one line function.

Reply:
this is from the f4 rcc.h